models/pose_estimation_mediapipe/demo.cpp

```diff
@@ -0,0 +1,2913 @@
+    Point2f pf(38.4, 40.7);
+    Point pi(pf);
+    cout << "Point float : " << pf << " "<< Mat(pf) << " "<< Mat(pf).rows << " "<< Mat(pf).cols<<endl;
+    cout << "Point int : " << pi << endl;
+    Mat xf = (Mat_<float>(2, 1) << 38.4, 40.7);
+    Mat xi;
+    xf.convertTo(xi, CV_32S);
+    cout << "Mat float : " << xf << endl;
+    cout << "Mat int : " << xi << endl;
+
+    parser.about("Person Detector from MediaPipe");
+    if (parser.has("help"))
+    {
+        parser.printMessage();
+        return 0;
+    }
+
+    string model = parser.get<String>("model");
+    float confThreshold = parser.get<float>("conf_threshold");; //  parser.get<float>("score_threshold");
+    float scoreThreshold = 0.5; //  parser.get<float>("score_threshold");
+    float nmsThreshold = 0.3; //  parser.get<float>("nms_threshold");
+    int topK = 5000; //  parser.get<int>("top_k");
+    bool vis = parser.get<bool>("vis");
+    bool save = parser.get<bool>("save");
+    int backendTargetid = parser.get<int>("backend");
+
+    if (model.empty())
+    {
+        CV_Error(Error::StsError, "Model file " + model + " not found");
+    }
+    VideoCapture cap;
+    if (parser.has("input"))
+        cap.open(samples::findFile(parser.get<String>("input")));
+    else
+        cap.open(0);
+    Mat frame;
+    // person detector
+    MPPersonDet modelNet("c:/users/laurent/Desktop/Nouveau dossier/person_detection_mediapipe_2023mar.onnx", nmsThreshold, scoreThreshold, topK,
+        backendTargetPairs[backendTargetid].first, backendTargetPairs[backendTargetid].second);
+    // pose estimator
+    MPPose poseEstimator(model, confThreshold, backendTargetPairs[backendTargetid].first, backendTargetPairs[backendTargetid].second);
+    //! [Open a video file or an image file or a camera stream]
+    if (!cap.isOpened())
+        CV_Error(Error::StsError, "Cannot opend video or file");
+
+    static const std::string kWinName = "MPPersonDet Demo";
+    int nbInference = 0;
+    while (waitKey() < 0)
+    {
+        cap >> frame;
+         if (frame.empty())
+        {
+            cout << "Frame is empty" << endl;
+            waitKey();
+            break;
+        }
+        TickMeter tm;
+        tm.start();
+        Mat person = modelNet.infer(frame);
+        tm.stop();
+        vector<tuple<Mat, Mat, Mat, Mat, Mat, float>> pose;
+        for (int idxRow = 0; idxRow < person.rows; idxRow++)
+        {
+            pose.push_back(poseEstimator.infer(frame, person.row(idxRow)));
+        }
+        cout << "Inference time: " << tm.getTimeMilli() << " ms\n";
+        pair<Mat, Mat> duoimg = visualize(frame, pose);
+        if (vis)
+        {
+            imshow(kWinName, get<0>(duoimg));
+            imshow("3d", get<1>(duoimg));
+        }
+        else
+        {
+            nbInference++;
+            if (nbInference > 100)
+            {
+                cout << nbInference << " inference made. Demo existing" << endl;
+                break;
+            }
```

Comment:
I think here we should do this like the previous PR, don't assume the 100 images.

Reply:
Ok